pkg/installer/installer.go

```diff
@@ -244,24 +299,97 @@ func NewDryRunInstaller(config *apiv1.KubectlStorageOSConfig) (*Installer, error
 		return installer, err
 	}
 
-	fileSys, err := buildInstallerFileSys(config, clientConfig)
+	distribution := pluginutils.DetermineDistribution(config.Spec.Install.KubernetesVersion)
+
+	installerOptions := &installerOptions{
+		storageosOperator: true,
+		storageosCluster:  !config.Spec.SkipStorageOSCluster,
+		portalClient:      config.Spec.Install.EnablePortalManager,
+		portalConfig:      config.Spec.Install.EnablePortalManager,
+		resourceQuota:     (distribution == pluginutils.DistributionGKE),
+		etcdOperator:      config.Spec.IncludeEtcd,
+		etcdCluster:       config.Spec.IncludeEtcd,
+	}
+
+	fileSys, err := installerOptions.buildInstallerFileSys(config, clientConfig)
 	if err != nil {
 		return installer, err
 	}
 
-	distribution := pluginutils.DetermineDistribution(config.Spec.Install.KubernetesVersion)
-
 	installer = &Installer{
-		distribution:  distribution,
-		clientConfig:  clientConfig,
-		stosConfig:    config,
-		fileSys:       fileSys,
-		onDiskFileSys: filesys.MakeFsOnDisk(),
+		distribution:     distribution,
+		clientConfig:     clientConfig,
+		stosConfig:       config,
+		fileSys:          fileSys,
+		onDiskFileSys:    filesys.MakeFsOnDisk(),
+		installerOptions: installerOptions,
 	}
 
 	return installer, nil
 }
 
+// NewUninstaller returns an Installer used for uninstall command
+func NewUninstaller(config *apiv1.KubectlStorageOSConfig) (*Installer, error) {
+	uninstaller := &Installer{}
+
+	clientConfig, err := pluginutils.NewClientConfig()
+	if err != nil {
+		return uninstaller, err
+	}
+
+	currentVersion, err := pluginutils.GetKubernetesVersion(clientConfig)
+	if err != nil {
+		return uninstaller, err
+	}
+
+	distribution := pluginutils.DetermineDistribution(currentVersion.String())
+
+	kubesystemNS, err := pluginutils.GetNamespace(clientConfig, "kube-system")
+	if err != nil {
+		return uninstaller, errors.WithStack(err)
+	}
+
+	uninstallPortal := false
```

Comment:
This should be initialized just before `if stosCluster != nil {` where it is actually used